corehq/apps/users/models.py

```diff
@@ -776,6 +776,8 @@ def wrap(cls, data, should_save=False):
         couch_user = super(CouchUser, cls).wrap(data)
         if should_save:
             couch_user.save()
+
+        couch_user._original_domains = set(couch_user.get_domains())
```

Comment:
also i just realized this is going to save _original_domains to the doc in couch which i suppose isn't a big deal, not sure of any way around it